client/extensions/wp-super-cache/state/stats/reducer.js

```diff
@@ -48,6 +53,38 @@ const generateStatus = createReducer( {}, {
 } );
 
 /**
+ * Returns the updated deleting state after an action has been dispatched.
+ * Deleting state tracks whether a cached file for a site is currently being deleted.
+ *
+ * @param  {Object} state Current deleting state
+ * @param  {Object} action Action object
+ * @return {Object} Updated deleting state
+ */
+const deleteStatus = createReducer( {}, {
+	[ WP_SUPER_CACHE_DELETE_FILE ]: ( state, { siteId } ) => ( {
+		...state,
+		[ siteId ]: {
+			deleting: true,
+			status: 'pending',
```

Comment:
IIRC I've seen this kind of reducer pattern in other parts of the WPSC code too. I think they could be simplified quite a bit (and concerns could be potentially better split) by solely storing a bool indicating status per `siteId` (as opposed to the object that currently also holds the `status` string field). (Right now, `status` is needed for the `hasDeleteError` selector, but [that could be dropped](https://github.com/Automattic/wp-calypso/pull/13818/files#r116734641).)